snuba/query/snql/__init__.py

```diff
@@ -1,28 +1,194 @@
+    """
+
+    def visit_query_exp(self, node, visited_children):
+        _, _, collect, _, _ = visited_children
+
+        return Query({}, None, collect, None,)
+
+    def visit_function_name(self, node: Node, visited_children: Iterable[Any]) -> str:
+        return visit_function_name(node, visited_children)
+
+    def visit_column_name(self, node: Node, visited_children: Iterable[Any]) -> Column:
+        return visit_column_name(node, visited_children)
+
+    def visit_low_pri_tuple(
+        self, node: Node, visited_children: Tuple[LowPriOperator, Expression]
+    ) -> LowPriTuple:
+        return visit_low_pri_tuple(node, visited_children)
+
+    def visit_high_pri_tuple(
+        self, node: Node, visited_children: Tuple[HighPriOperator, Expression]
+    ) -> HighPriTuple:
+        return visit_high_pri_tuple(node, visited_children)
+
+    def visit_low_pri_op(
+        self, node: Node, visited_children: Iterable[Any]
+    ) -> LowPriOperator:
+        return visit_low_pri_op(node, visited_children)
+
+    def visit_high_pri_op(
+        self, node: Node, visited_children: Iterable[Any]
+    ) -> HighPriOperator:
+        return visit_high_pri_op(node, visited_children)
+
+    def visit_arithmetic_term(
+        self, node: Node, visited_children: Tuple[Any, Expression, Any]
+    ) -> Expression:
+        return visit_arithmetic_term(node, visited_children)
+
+    def visit_low_pri_arithmetic(
+        self,
+        node: Node,
+        visited_children: Tuple[Any, Expression, Any, LowPriArithmetic],
+    ) -> Expression:
+        return visit_low_pri_arithmetic(node, visited_children)
+
+    def visit_high_pri_arithmetic(
+        self,
+        node: Node,
+        visited_children: Tuple[Any, Expression, Any, HighPriArithmetic],
+    ) -> Expression:
+        return visit_high_pri_arithmetic(node, visited_children)
+
+    def visit_numeric_literal(
+        self, node: Node, visited_children: Iterable[Any]
+    ) -> Literal:
+        return visit_numeric_literal(node, visited_children)
+
+    def visit_quoted_literal(
+        self, node: Node, visited_children: Tuple[Any, Node, Any]
+    ) -> Literal:
+        return visit_quoted_literal(node, visited_children)
+
+    def visit_collect_clause(self, node, visited_children):
+        _, _, selected_columns, _ = visited_children
+        return selected_columns
+
+    def visit_collect_columns(self, node, visited_children):
+        columns, _, _, _, = visited_children
+        return columns
+
+    def visit_collect_list(self, node, visited_children):
+        column_list, right_column = visited_children
+        ret: List[SelectedExpression] = []
+        if not isinstance(column_list, Node):
+            if not isinstance(column_list, (list, tuple)):
+                ret.append(SelectedExpression(None, column_list))
+            else:
+                for p in column_list:
+                    ret.append(SelectedExpression(None, p))
```

Comment:
This is missing one part: you are not assigning the names to the SerlectedExpression objects. SelectedExpression.name should be the name the user expects in the output. It works like for SQL:
```
SELECT a, b, f(x), g(k) AS alias....
```
Would look like in the output
```
a | b | f(x) | alias
------------------
values. ....
```

So the algorithm should work as:
1) if the parsed expression (column_list at line 152 and p here) has an alias then the name is the alias.
2) if not, you should use the string the user provided, which you need to preserve the raw expression, which should be the raw string for low_pri_aritthmetic from the `collect_columns       = low_pri_arithmetic space* comma space*` rule. The raw string should be accessible from the parsed node.